templates/common/baremetal/files/NetworkManager-static-dhcp.yaml

```diff
@@ -0,0 +1,30 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/30-static-dhcp"
+contents:
+  inline: |
+    #!/bin/bash
+    set -ex
+
+    if [ ${DHCP4_DHCP_LEASE_TIME:-0} -lt 4294967295 ]
+    then
+        echo "Not an infinite lease. Ignoring."
+        exit 0
+    fi
+
+    IPS=($IP4_ADDRESS_0)
+    CIDR=${IPS[0]}
+    GATEWAY=${IPS[1]}
```

Comment:
Is there some way to retrieve all of the dhcp configuration data and easily add it? Things like hostname would get probably persisted by our other dispatcher script, but we may be missing things like the search domain.

Reply:
Maybe using dbus?

Reply:
Getting the active connections
```
busctl get-property --json=pretty org.freedesktop.NetworkManager /org/freedesktop/NetworkManager org.freedesktop.NetworkManager ActiveConnections
{
        "type" : "ao",
        "data" : [
                "/org/freedesktop/NetworkManager/ActiveConnection/802",
                "/org/freedesktop/NetworkManager/ActiveConnection/2"
        ]
}
```

Checking the name (which I suspect is what we get from dispatcher scripts)
```
❯ busctl get-property --json=pretty org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/ActiveConnection/802 org.freedesktop.NetworkManager.Connection.Active Id
{
        "type" : "s",
        "data" : "MYWIFI"
}
```

Retrieving the DHCP4 object path:
```
busctl get-property --json=pretty org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/ActiveConnection/802 org.freedesktop.NetworkManager.Connection.Active Dhcp4Config
{
        "type" : "o",
        "data" : "/org/freedesktop/NetworkManager/DHCP4Config/821"
}
```

Retrieving the DHCPv4 properties
```
busctl get-property --json=pretty org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/DHCP4Config/821 org.freedesktop.NetworkManager.DHCP4Config Options
{
	"type" : "a{sv}",
	"data" : {
		"dhcp_lease_time" : {
			"type" : "s",
			"data" : "3600"
		},
		"domain_name" : {
			"type" : "s",
			"data" : "home.myhome.cz"
		},
		"domain_name_servers" : {
			"type" : "s",
			"data" : "192.168.2.1"
		},
		"domain_search" : {
			"type" : "s",
			"data" : "home.myhome.cz"
		},
		"expiry" : {
			"type" : "s",
			"data" : "1603931004"
		},
		"host_name" : {
			"type" : "s",
			"data" : "bucek.home.myhome.cz"
		},
		"ip_address" : {
			"type" : "s",
			"data" : "192.168.2.37"
		},
		"requested_broadcast_address" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_domain_name" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_domain_name_servers" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_domain_search" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_host_name" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_interface_mtu" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_ms_classless_static_routes" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_nis_domain" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_nis_servers" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_ntp_servers" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_rfc3442_classless_static_routes" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_root_path" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_routers" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_static_routes" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_subnet_mask" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_time_offset" : {
			"type" : "s",
			"data" : "1"
		},
		"requested_wpad" : {
			"type" : "s",
			"data" : "1"
		},
		"routers" : {
			"type" : "s",
			"data" : "192.168.2.1"
		},
		"subnet_mask" : {
			"type" : "s",
			"data" : "255.255.255.0"
		}
	}
}
```

Reply:
Hmm, I'll have to look into this. I don't see the search domain in the dispatcher env vars, but I also didn't specify one on the DHCP server side so it may just not have been populated.

Reply:
If the DHCP server sets a search domain it will be available in the dispatcher env vars. We just need to map it to nmcli, which I will do in my next update to this. If there's anything else you think we should capture, let me know.